### ISSUE TITLE
Have `/decide` include disabled flag keys mapping to `null`

### ISSUE THREAD

**Issue by havenbarnes** [PostHog/posthog#26948]
Our docs mention that disabled flags will return as either `null` or `undefined` (https://posthog.com/docs/feature-flags/creating-feature-flags#enable-feature-flag). In reality `/decide` will simply not include these keys at all (which is technically equivalent to `undefined` after HTTP transport). It would be more useful for clients to see these keys mapped to `null` every time, so that SDK users can differentiate between a disabled flag and non-existent flag / posthog error response.

**Comment by benjackwhite** [comment]
One thing to consider - people might purposefully want it this way. So that they don't show flag keys that the user shouldn't have access to. 

I can't think of a super strong reason why but simply showing all keys might not be what's wanted 🤷

Also many older SDKs don't check the value, they just grab the keys instead. Would need to be sure to make this a version change to the api (or maybe even just make this part of the new flags service instead)

**Comment by haacked** [comment]
> One thing to consider - people might purposefully want it this way. So that they don't show flag keys that the user shouldn't have access to.

Yeah, #24994 brings up a good point. Suppose I have an upcoming feature `beta-mind-reading-mode` that we want to keep secret until it's ready. Anyone can inspect their `/decide` request and see all the flags.

Perhaps we should keep the default to omit these keys. But we could add a `show_all` parameter to the query string that would show all the flags and their results. This would be useful during development and debugging.

**Comment by haacked** [comment]
Ok, sorry to spam this issue, but something just occurred to me that I wrote in another issue:

> As I type this, it occurs to me, the only way a user can snoop on these requests is if they're made in the browser. Even if we omit the feature flag keys, a user can still see the code calling it:
> So not sure omitting these keys actually solves a whole lot.

So unless I'm missing something, I'm not sure what omitting the keys would solve.

**Comment by haacked** [comment]
Note that this issue is in direct contradiction to #24994. So we have to make a judgment call here. We could add a query string parameter to control the behavior. Or we can close #24994 and do this. Or vice versa. I think the key here is to make a decision.

**Comment by haacked** [comment]
The more I think about it, I think we should close this issue. I think omitting flags that are disabled is the current behavior and is the correct behavior. If you want the flag to be disabled but included in the payload, enable it and set its rollout to 0%.
